clang-tools-extra/test/clang-tidy/checkers/bsl-non-pod-classdef.cpp

```diff
@@ -0,0 +1,34 @@
+// RUN: %check_clang_tidy %s bsl-non-pod-classdef %t
```

Comment:
I think you have more than one check here. One for POD and one for the loop stuff. I would break these up into their own checks.

Reply:
Not sure I follow - this does have two checks: 
Non-POD member data should be private
Non-POD type should be defined as class
since I thought they were similar, but the for loop stuff is the other check. Do you still want these two broken up?

Reply:
ah I initially had the two checks on the same branch and tried to cherry pick them to separate into two branches, but it wasn't super clean, so I was going to merge and pull the for-loop PR first so those should go away before I merge this one.